src/main/java/core/basesyntax/ArrayList.java

```diff
@@ -1,49 +1,120 @@
 package core.basesyntax;
 
+import java.util.NoSuchElementException;
+import java.util.Objects;
+
 public class ArrayList<T> implements List<T> {
+    private static final int DEFAULT_ARRAY_SIZE = 10;
+    private Object[] array;
+    private int currentSize;
+
+    public ArrayList() {
+        currentSize = 0;
+        this.array = new Object[DEFAULT_ARRAY_SIZE];
+    }
+
+    private Object[] grow(Object[] oldArray) {
+        Object[] grownArray = new Object [oldArray.length + (oldArray.length >> 1)];
+        System.arraycopy(oldArray, 0, grownArray, 0, currentSize);
+        return grownArray;
+    }
 
     @Override
     public void add(T value) {
+        if (currentSize < array.length) {
+            array[currentSize] = value;
+            currentSize++;
+        } else {
+            array = grow(array);
+            this.add(value);
```

Comment:
let's refactor this so that we won't use recursion

also, don't use `this` keyword if it is not necessary